main.py

```diff
@@ -0,0 +1,69 @@
+"""Proxy server to communicate a client to ARTIQ."""
+
+from contextlib import asynccontextmanager
+import json
+import os
+
+from fastapi import FastAPI
+from sipyco.pc_rpc import Client
+
+
+settings = {}
+
+
+def load_setup_file():
+    """Loads set-up information from the setup file.
+
+    The file should have the following JSON structure:
+
+      {
+        "master_path": {master_path}
+      }
+    """
+    global settings
+    with open("setup.json", encoding="utf-8") as setup_file:
+        settings = json.load(setup_file)
+
+
+@asynccontextmanager
+async def lifespan(_app: FastAPI):
+    """Lifespan events.
+
+    This function is set as the lifespan of the application.
+
+    The code before 'yield' is executed once before the application starts.
+    The code after 'yield' is executed when the application is shutting down.
```

Comment:
Even though it was helpful for me to understand the code, but I think we don't have to explain this in the code.
Since we use this explicitly `app = FastAPI(lifespan=lifespan)`, anyone who does not know about the lifespan can read the FastAPI documentation.

When you write comments, you may exclude the general information about the API or library. Assume that the reader knows the library as well as you do. Such information can be given in the PR, and it could help the reviews a lot, as you did 👍